app/views/protocols_participants/_appointments.html.haml

```diff
@@ -35,7 +35,7 @@
         = f.label :arm_id, t(:appointments)[:arm_change_label]
         = f.select :arm_id, protocols_participant_calendar_arm_options(protocols_participant), {}, class: 'selectpicker', id: "change_arm_dropdown", onchange: "Rails.fire(this.form, 'submit')"
     .list-group.list-group-flush
-      - appointments.each do |appt|
+      - appointments.sort_by{|appt| appt.visit_group.day}.each do |appt|
```

Comment:
Appointments can exist without a visit_group (custom appointments added just for one participant), so this will throw an error in that instance. (I believe that's why the specs are failing).

Instead, you can use the "position" attribute of the appointment itself. That will always be present, and should be the same order.